ts/src/profiler.ts

```diff
@@ -81,16 +84,30 @@ function getServerResponseBackoff(response: http.IncomingMessage): number|
     undefined {
   // tslint:disable-next-line: no-any
   const body = (response as any).body;
-  if (body && body.error && body.error.details &&
-      Array.isArray(body.error.details)) {
-    for (const item of body.error.details) {
-      if (typeof item === 'object' && item.retryDelay &&
-          typeof item.retryDelay === 'string') {
-        const backoffMillis = parseDuration(item.retryDelay);
-        if (backoffMillis > 0) {
-          return backoffMillis;
-        }
-      }
+
+  // The response currently does not have field containing the server-specified
+  // backoff. As a workaround, response body's message is parsed to get the
+  // backoff.
+  // TODO (issue #250): Remove this workaround and get the retry delay from
+  // body.error.details.
+  if (body && body.message && typeof body.message === 'string') {
+    return parseBackoffDuration(body.message);
+  }
+  return undefined;
+}
+
+/**
+ * @return if the backoff duration can be parsed, then the backoff duration in
+ * ms, otherwise undefined.
+ *
+ * Public for testing.
+ */
+export function parseBackoffDuration(backoffStr: string): number|undefined {
+  const found = backoffStr.match(BACKOFF_MSG_PAT);
+  if (found && found.length >= 2 && typeof found[1] === 'string') {
```

Comment:
You could write this as follows. IMO this is more readable.

```ts
const [, duration] = BACKOFF_MSG_PATH.exec(backoffStr);
```

Other nits:
* The parameter `parameterStr` include the (abbreviated) type in the name. It would be better to call it `backoffMessage` or just `backoff`.
* Personal opinion: the regex being declared at the top of the file makes it harder to understand what this code is doing. I would suggest putting the regex (or the variable) in this function.

Reply:
Putting the regexp var within the function would make it instantiate on every call?

Reply:
For now, I moved the regex closer to the function, but I'd be happy to put the regex in the function if it won't instantiate on every call.

Reply:
I would call this premature optimization :p.

There is marginal difference in performance: http://jsben.ch/jsLlz.

Reply:
Actually there is no statistically significant difference in performance.

Reply:
Done.